qemu/tests/hpt_max_page_size.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+from virttest import env_process
+from virttest import error_context
+from virttest.staging import utils_memory
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test to maximum page size for hash page table guests.
+    Steps:
+    1) There are two options, with men backing file or without men backing
+       file, if without men backing, ignore the step 3.
+    2) System setup hugepages on host.
+    3) Mount this hugepage to /mnt/kvm_hugepage.
+    4) Boot up guest with cap-hpt-max-page-size.
+    5) Check the hugepage value is correct inside guest.
+
+    :params test: QEMU test object.
+    :params params: Dictionary with the test parameters.
+    :params env: Dictionary with test environment.
+    """
+    def _check_hugepages_size():
+        """ Check hugepages of vm. """
+        for hugepage_size in session.cmd_output(
+                params["hp_size_cmd"]).split('\n')[:2]:
```

Comment:
Better to use `utils_memory.get_num_anon_huge_pages(session)` to get `AnonHugePages`, and use `utils_memory.read_from_meminfo("ShmemHugePages", session)` to get `ShmemHugePages` if you need.

Reply:
the get_num_anon_huge_pages(session) ,with 'grep %s /proc/%s/smaps'. the value different from ‘/proc/meminfo’.